core/src/main/java/org/apache/spark/shuffle/sort/ShuffleExternalSorter.java

```diff
@@ -161,6 +161,10 @@ private void writeSortedFile(boolean isLastFile) {
     final ShuffleInMemorySorter.ShuffleSorterIterator sortedRecords =
       inMemSorter.getSortedIterator();
 
+    // If there are no sorted records, so we don't need to create an empty spill file.
+    if (!sortedRecords.hasNext()) {
+      return;
+    }
```

Comment:
@wangjiaochun . Can we move this line 160 ~ 167 to line 147? If then, we don't need to do `new SuffleWriteMetrics()` at all.